tests/unit/test_parallel_write.py

```diff
@@ -0,0 +1,185 @@
+
+    def _get_data(self, selection: Tuple[slice]) -> np.ndarray:
+        return self.data[selection]
+    
+def test_parallel_write(tmpdir):
+    number_of_jobs = 2
+    data = np.array([1., 2., 3.])
+    column = VectorData(name="TestColumn", description="", data=PickleableDataChunkIterator(data=data))
+    dynamic_table = DynamicTable(name="TestTable", description="", columns=[column])
+
+    zarr_top_level_path = str(tmpdir / "test_parallel_write.zarr")
+    with ZarrIO(path=zarr_top_level_path,  manager=get_manager(), mode="w") as io:
+        io.write(dynamic_table, number_of_jobs=number_of_jobs)
+
+    # TODO: roundtrip currently fails due to read error
+    #with ZarrIO(path=zarr_top_level_path, mode="r") as io:
+    #    dynamic_table_roundtrip = io.read()
+    #    data_roundtrip = dynamic_table_roundtrip["TestColumn"].data
+    #    assert_array_equal(data_roundtrip, data)
+        
+        
+def test_mixed_iterator_types(tmpdir):
+    number_of_jobs = 2
+    generic_column = VectorData(name="TestGenericColumn", description="", data=PickleableDataChunkIterator(data=np.array([1., 2., 3.])))
+    classic_column = VectorData(name="TestClassicColumn", description="", data=DataChunkIterator(data=np.array([4., 5., 6.])))
+    unwrapped_column = VectorData(name="TestUnwrappedColumn", description="", data=np.array([7., 8., 9.]))
+    dynamic_table = DynamicTable(name="TestTable", description="", columns=[generic_column, classic_column, unwrapped_column])
+
+    zarr_top_level_path = str(tmpdir / "test_mixed_iterator_types.zarr")
+    with ZarrIO(path=zarr_top_level_path,  manager=get_manager(), mode="w") as io:
+        io.write(dynamic_table, number_of_jobs=number_of_jobs)
+        
+    # TODO: roundtrip currently fails 
+    #with ZarrIO(path=zarr_top_level_path, mode="r") as io:
+    #    dynamic_table_roundtrip = io.read()
+    #    data_roundtrip = dynamic_table_roundtrip["TestColumn"].data
+    #    assert_array_equal(data_roundtrip, data)
+
+def test_mixed_iterator_pickleability(tmpdir):
+    number_of_jobs = 2
+    pickleable_column = VectorData(name="TestGenericColumn", description="", data=PickleableDataChunkIterator(data=np.array([1., 2., 3.])))
+    not_pickleable_column = VectorData(name="TestClassicColumn", description="", data=NotPickleableDataChunkIterator(data=np.array([4., 5., 6.])))
+    dynamic_table = DynamicTable(name="TestTable", description="", columns=[pickleable_column, not_pickleable_column])
+
+    zarr_top_level_path = str(tmpdir / "test_mixed_iterator_pickleability.zarr")
+    with ZarrIO(path=zarr_top_level_path,  manager=get_manager(), mode="w") as io:
+        io.write(dynamic_table, number_of_jobs=number_of_jobs)
+
+    # TODO: roundtrip currently fails due to read error
+    #with ZarrIO(path=zarr_top_level_path, mode="r") as io:
+    #    dynamic_table_roundtrip = io.read()
+    #    data_roundtrip = dynamic_table_roundtrip["TestColumn"].data
+    #    assert_array_equal(data_roundtrip, data)
```

Comment:
see #116 for blocking issue including this extra layer to the tests here

Reply:
I added some thoughts on issue #116. Long story short, I think if you modify line 115 to set the `id` parameter correctly, then this should work (I think).